sql/core/src/main/scala/org/apache/spark/sql/execution/AliasAwareOutputExpression.scala

```diff
@@ -48,13 +42,13 @@ trait AliasAwareOutputExpression extends UnaryExecNode {
  */
 trait AliasAwareOutputPartitioning extends AliasAwareOutputExpression {
   final override def outputPartitioning: Partitioning = {
-    if (hasAlias) {
-      child.outputPartitioning match {
-        case h: HashPartitioning => h.copy(expressions = replaceAliases(h.expressions))
-        case other => other
-      }
-    } else {
-      child.outputPartitioning
+    child.outputPartitioning match {
+      case e: Expression if hasAlias =>
+        val normalizedExp = e.transformDown {
```

Comment:
nit: `normalizedExp` -> `normalizedExpr` and `transformDown` -> `transform`

Reply:
btw, we need `transform` here?

Reply:
renaming -> done.

yes transform is needed here as each Partitioning can be a tree in itself. Example for INNER join, output partitioning is: PartitioningCollection with 2 HashPartitioning as childrens. Also in each HashPartitioning you can have expressions like 2*X, which might needs to be transformed to 2*aliasedX.